internal/commands/image/tag.go

```diff
@@ -0,0 +1,85 @@
+package image
+
+import (
+	"fmt"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/app/internal/store"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/distribution/reference"
+	"github.com/spf13/cobra"
+)
+
+func tagCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Short: "Create a new tag from an application image",
+		Use:   "tag SOURCE_APP_IMAGE[:TAG] TARGET_APP_IMAGE[:TAG]",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			appstore, err := store.NewApplicationStore(config.Dir())
+			if err != nil {
+				return err
+			}
+
+			bundleStore, err := appstore.BundleStore()
+			if err != nil {
+				return err
+			}
+
+			if len(args) != 2 {
+				return fmt.Errorf("\"docker image tag\" requires exactly 2 arguments.\nSee 'docker app image tag --help'")
+			}
+
+			return runTag(bundleStore, args[0], args[1])
+		},
+	}
+
+	return cmd
+}
+
+func runTag(bundleStore store.BundleStore, srcAppImage, destAppImage string) error {
+	srcRef, err := readBundle(srcAppImage, bundleStore)
+	if err != nil {
+		return err
+	}
+
+	if err = storeBundle(srcRef, destAppImage, bundleStore); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func readBundle(name string, bundleStore store.BundleStore) (*bundle.Bundle, error) {
+	cnabRef, err := stringToRef(name)
+	if err != nil {
+		return nil, err
+	}
+
+	bundle, err := bundleStore.Read(cnabRef)
```

Comment:
You can just `return bundleStore.Read(cnabRef)` this will remove the redundant error message: `cound not read bundle: <NAME>: failed to read bundle <REF>`